src/mame/konami/thunderx.cpp

```diff
@@ -444,10 +388,44 @@ void thunderx_state::calculate_collisions()
 		s1 = m_pmcram[6];
 	}
 
-	const int cm = m_pmcram[3];
-	const int hm = m_pmcram[4];
+	const uint8_t cm = m_pmcram[3];
+	const uint8_t hm = m_pmcram[4];
+
+	// collide objects from s0 to e0 against objects from s1 to e1
+	// only process objects with the specified bits (cm/hm) set in their flags
+	//
+	// the data format for each object is:
+	//
+	// +0 : flags
+	// +1 : width (4 pixel units)
+	// +2 : height (4 pixel units)
+	// +3 : x (2 pixel units) of center of object
+	// +4 : y (2 pixel units) of center of object
+	for (uint8_t *p0 = &m_pmcram[s0]; p0 < &m_pmcram[e0]; p0 += 5)
+	{
+		// check object 0 flags
+		if (!(p0[0] & cm))
+			continue;
+
+		for (uint8_t *p1 = &m_pmcram[s1]; p1 < &m_pmcram[e1]; p1 += 5)
+		{
+			// check object 1 flags
+			if (!(p1[0] & hm))
+				continue;
+			
+			if (p1[1] + p0[1] < abs(p1[3] - p0[3])) continue;
+			if (p1[2] + p0[2] < abs(p1[4] - p0[4])) continue;
 
-	run_collisions(s0, e0, s1, e1, cm, hm);
+			// set flags
+			p1[0] = (p1[0] & 0x8f) | 0x10;
+			if (p1 > (uint8_t*)0xe6)	// This address value is hardcoded in the PMC program
```

Comment:
@furrtek this pointer comparison won't work since &m_pmcram[0] is not 0.
If I read your disasm correctly, it's (&p1[4] >= &m_pmcram[0xe6]), but then collisions don't work anymore.

Reply:
Maybe p0[4] instead of p1[4], but then I see another issue where the address value in the US version is $36 instead of $E6,